### ISSUE TITLE
Add PWideChar type

### ISSUE THREAD

**Issue by dezlov** [remobjects/pascalscript#110]
Please add `PWideChar` type to allow using DLL functions which expect null-terminated Unicode strings.

Many API calls (particularly WinAPI) use `PWideChar` parameters (alias of `LPCWSTR` type in Delphi/FPC) when dealing with Unicode strings. Using such functions is currently not possible at all in PascalScript, as far as I know.

**Comment by pult** [comment]
PWideChar https://github.com/pult/pascalscript

**Comment by dezlov** [comment]
> PWideChar https://github.com/pult/pascalscript

Thanks for the link.

I tried to look though your commit history, but it's hard to follow. There seems to be many modifications to various files, committed on per file bases without meaningful commit message.

Any chance you could isolate the `PWideChar` integration, so it could potentially be pulled into *PascalScript*?

**Comment by pult** [comment]
You need to look "btPWideChar / PWideChar". All modifications are marked as "{+} ... {+.}". Besides "WideChar" More added more complete support "interfaces" (I had to refer to the wmi).

**Comment by martijnlaan** [comment]
Assuming you're using a Unicode compiler, sending Unicode strings to DLLs already works fine, just use `String` as the parameter type.

**Comment by dezlov** [comment]
> Assuming you're using a Unicode compiler...

I'm using FPC where `String` type is `AnsiString`.

PascalScript should support both PAnsiChar and PWideChar, irrespective of the Unicode-ness of compiler.
